internal/controller/utils.go

```diff
@@ -77,6 +78,66 @@ func reconcileStatefulSet(ctx context.Context, logger logr.Logger, ec *ecv1alpha
 	return getStatefulSet(ctx, c, ec.Name, ec.Namespace)
 }
 
+func setDefaultArgs(name string) []string {
+	return []string{
+		"--name=$(POD_NAME)",
+		"--listen-peer-urls=http://0.0.0.0:2380",   // TODO: only listen on 127.0.0.1 and host IP
+		"--listen-client-urls=http://0.0.0.0:2379", // TODO: only listen on 127.0.0.1 and host IP
+		fmt.Sprintf("--initial-advertise-peer-urls=http://$(POD_NAME).%s.$(POD_NAMESPACE).svc.cluster.local:2380", name),
+		fmt.Sprintf("--advertise-client-urls=http://$(POD_NAME).%s.$(POD_NAMESPACE).svc.cluster.local:2379", name),
+	}
+}
+
+func RemoveStringFromSlice(defaultArgs []string, s string) []string {
+	var compareStr string
+	var index int
+	exists := false
+
+	for i := range defaultArgs {
+		compareStr = getArgName(defaultArgs[i])
+		if compareStr == s {
+			index = i
+			exists = true
+			break
+		}
+	}
+	if exists {
+		defaultArgs = slices.Delete(defaultArgs, index, index+1)
+	}
+	return defaultArgs
+}
+
+func getArgName(s string) string {
+	idx := strings.Index(s, "=")
+
+	if idx != -1 {
+		return s[:idx]
```

Comment:
Please try to remove the leading and trailing space for the returned string using `strings.TrimSpace`